rotkehlchen/exchanges/coinbase.py

```diff
@@ -220,27 +327,24 @@ def _validate_single_api_key_action(
 
         return result, ''
 
-    def validate_api_key(self) -> tuple[bool, str]:
-        """Validates that the Coinbase API key is good for usage in rotki
+    def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> bool:
+        changed = super().edit_exchange_credentials(credentials)
+        if credentials.api_key is not None:
+            try:
+                new_is_legacy = self.is_legacy_key(credentials.api_key)
+            except AuthenticationError as e:
+                log.error(f'Invalid coinbase API key format: {e}')
 
-        Makes sure that the following permissions are given to the key:
-        wallet:accounts:read, wallet:transactions:read,
-        wallet:withdrawals:read, wallet:deposits:read
-        """
-        result, msg = self._validate_single_api_key_action('accounts')
-        if result is None:
-            return False, msg
-
-        # now get the account ids
-        account_info = self._get_active_account_info(result)
-        if len(account_info) != 0:
-            # and now try to get all transactions of an account to see if that's possible
-            method = f'accounts/{account_info[0][0]}/transactions'
-            result, msg = self._validate_single_api_key_action(method)
-            if result is None:
-                return False, msg
+            if new_is_legacy != self.is_legacy_api_key:  # Key type has changed
+                self.is_legacy_api_key = new_is_legacy
+                changed = True
```

Comment:
This is still here @vijithreddy. Please remove as asked here: https://github.com/rotki/rotki/pull/7903/files#r1596587452
```suggestion
```